modules/scalacheck/shared/src/main/scala/eu/timepit/refined/scalacheck/numeric.scala

```diff
@@ -1,82 +1,75 @@
 package eu.timepit.refined
 package scalacheck
 
-import eu.timepit.refined.api.{RefType, Validate}
+import eu.timepit.refined.api._
 import eu.timepit.refined.numeric._
 import eu.timepit.refined.scalacheck.util.{Adjacent, Bounded}
 import org.scalacheck.{Arbitrary, Gen}
 import org.scalacheck.Gen.Choose
 import shapeless.{Nat, Witness}
 import shapeless.ops.nat.ToInt
+import shapeless.tag.@@
 
 /**
  * Module that provides `Arbitrary` instances and generators for
  * numeric predicates.
  */
 object numeric extends NumericInstances
 
-trait NumericInstances {
-
-  /**
-   * A generator that generates a random value in the given (inclusive)
-   * range that satisfies the predicate `P`. If the range is invalid,
-   * the generator will not generate any value.
-   *
-   * This is like ScalaCheck's `Gen.chooseNum` but for refined types.
-   */
-  def chooseRefinedNum[F[_, _], T: Numeric: Choose, P](min: F[T, P], max: F[T, P])(
-      implicit rt: RefType[F],
-      v: Validate[T, P]
-  ): Gen[F[T, P]] =
-    Gen.chooseNum(rt.unwrap(min), rt.unwrap(max)).filter(v.isValid).map(rt.unsafeWrap)
-
-  ///
-
+trait DeprecatedNumericInstances {
+  @deprecated("Use numericMaxAndMinArbitrary instead", "0.8.6")
```

Comment:
I've tried to keep the old arbitrary instances for backwards compat - sadly this only seems to have worked on Scala 2.11. Any good ideas how to make this work?